### PR TITLE
Cleans up sleeping carp exclusive code in human_defense.dm

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

See title. Adds a hook for on for on projectile hit for martial arts.

## Why It's Good For The Game

Allows for easier creation of new martial arts with other projectile effects. Gets rid of dead code. Moves code to where it should actually be. Overall a lot saner and nicer looking.

Code should almost always have the same results, except in some extreme edge cases, so no changelog needed.

### CODE REVIEW

**Comment by nemvar** [comment]
RIP varediting admins though.

**Review by actioninja — DISMISSED** [review]

File: code/datums/martial/sleeping_carp.dm [hunk 1]
```diff
@@ -148,6 +146,23 @@
...
+/datum/martial_art/the_sleeping_carp/on_projectile_hit(mob/living/carbon/human/A, obj/item/projectile/P, def_zone)
+	. = ..()
+	if(A.incapacitated(FALSE, TRUE)) //NO STUN
+		return
+	if(!(A.mobility_flags & MOBILITY_USE)) //NO UNABLE TO USE
+		return
+	if(A.dna && A.dna.check_mutation(HULK)) //NO HULK
+		return
+	if(!isturf(A.loc)) //NO MOTHERFLIPPIN MECHS!
+		return
```
Comment (SpaceManiac) [inline code/datums/martial/sleeping_carp.dm]: Are these supposed to be returning null rather than `BULLET_ACT_HIT`?

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: code/datums/martial/sleeping_carp.dm (hunk 1 above)
Comment (nemvar) [reply code/datums/martial/sleeping_carp.dm]: They return what the parent returns. Which is BULLET_ACT_HIT
```
/datum/martial_art/proc/on_projectile_hit(mob/living/carbon/human/A, obj/item/projectile/P, def_zone)
	return BULLET_ACT_HIT

Comment (nemvar) [reply code/datums/martial/sleeping_carp.dm]: Guess I could also add that there. Wouldn't change any functionality, but does look a bit better.

Comment (SpaceManiac) [reply code/datums/martial/sleeping_carp.dm]: Ah, probably fine then

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**